apps/docs/src/guide/predicates/send-and-spend-funds-from-predicates.md

```diff
@@ -40,26 +40,20 @@ Once the predicate resolves with a return value `true` based on its predefined c
 
 In a similar approach, you can use the `createTransfer` method, which returns a [`ScriptTransactionRequest`](../../api/Account/ScriptTransactionRequest.md). Then, we can submit this transaction request by calling the `sendTransaction` method.
 
-<<< @/../../docs-snippets/src/guide/predicates/send-and-spend-funds-from-predicates.test.ts#send-and-spend-funds-from-predicates-8{ts:line-numbers}
+The following example, we are pre-staging a transaction and therefore we are able to know the transaction ID without actually submitting the transaction.
+
+<<< @/../../docs-snippets2/src/predicates/cookbook/pre-stage.ts#send-and-spend-funds-from-predicates-8{ts:line-numbers}
 
 ## Spending Entire Predicate Held Amount
 
 Trying to forward the entire amount held by the predicate results in an error because no funds are left to cover the transaction fees. Attempting this will result in an error message like:
 
-<<< @/../../docs-snippets/src/guide/predicates/send-and-spend-funds-from-predicates.test.ts#send-and-spend-funds-from-predicates-6{ts:line-numbers}
+<<< @/../../docs-snippets2/src/predicates/cookbook/failure-not-enough-funds.ts#send-and-spend-funds-from-predicates-6{ts:line-numbers}
 
 ## Predicate Validation Failure
 
 What happens when a predicate fails to validate? Recall our predicate only validates if the `input_address` matches the hard-coded `valid_address`. Hence, if we set a different data from the `valid_address`, the predicate will fail to validate.
 
 When a predicate fails to validate, the SDK throws an error that starts like this:
 
-<<< @/../../docs-snippets/src/guide/predicates/send-and-spend-funds-from-predicates.test.ts#send-and-spend-funds-from-predicates-7{ts:line-numbers}
-
-## Pre-staging a Transaction
-
-In some cases, you may want to pre-stage a predicate transaction before submitting it for execution. To do this, you can use the `createTransfer` method on the `Predicate` class.
-
-In the following example, we are pre-staging a transaction to be able to know the transaction ID without actually submitting the transaction.
-
-<<< @/../../docs-snippets/src/guide/predicates/send-and-spend-funds-from-predicates.test.ts#predicates-prestage-transaction{ts:line-numbers}
+<<< @/../../docs-snippets2/src/predicates/cookbook/failure-returns-false.ts#send-and-spend-funds-from-predicates-7{ts:line-numbers}
```

Comment:
Did you remove the "Pre-stating a Transaction" guide section on purpose?

Reply:
It was almost identical to the [snippet above](https://github.com/FuelLabs/fuels-ts/pull/3358/files#diff-c91f563a89b2a75ace2424867818c2a79dbcb2fabc073472c3ddfebcc370f79aR41-R45), so I move the relevant information.